jax/random.py

```diff
@@ -261,8 +291,8 @@ def fold_in(key, data):
 
 @jit
 def _fold_in(key, data):
-  key2 = lax.tie_in(key, PRNGKey(data))
-  return threefry_2x32(key, key2)
+  key2 = lax.tie_in(key, np.asarray([[0, data]], dtype='uint32'))
```

Comment:
Instead of always padding left with zeros, we could make this function handle 64bit values like you do elsewhere. Maybe leave a TODO?

I think concatenating a zero with lax operations would be better than using `np.asarray` here, since `np.asarray` can be much more complex. Under a jit it's okay, but it's kind of a bad style for others to copy.

Reply:
I've reverted to feeding `data` into `PRNGKey`, so that higher order bits are handled, as they were before (I got thrown by the docstring which previously said that fold_in only handled 32 bit ints).